duckduckgo/BrowserServicesKit
# Extract SecureVault code into SecureStorage module

Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/0/1205119749390492/f
iOS PR: TODO
macOS PR: https://github.com/duckduckgo/macos-browser/pull/1382
What kind of version bump will this require?: Major

**Optional**:

Tech Design URL:
CC: @Bunn @jotaemepereira 

**Description**:

This PR extracts the reusable SecureVault logic out into a new `SecureStorage` module.

This module is used by Autofill, and is usable by the client apps directly to implement secure storage on their own. The API for this new module matches the existing API as much as possible, and the implementation of the key/crypto providers is made easier by providing default implementations of the generic functions.

Note that not all areas of the code have been renamed; for example, there is still a struct called `SecureVaultModels` which I would like to rename `AutofillModels`, but in this case this symbol is referenced over 600 times, so I plan to do this in a follow-up renaming PR to keep this PR as small as possible (it's already quite large).



**Steps to test this PR**:
1. Make sure that tests pass
1. Run the client PRs and make sure that Autofill functionality all works as expected



**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
